pkg/controller/datavolume/controller.go

```diff
@@ -236,63 +231,81 @@ func getDataVolumeOp(dv *cdiv1.DataVolume) dataVolumeOp {
 	return dataVolumeNop
 }
 
-// Reconcile loop for the data volumes
-func (r ReconcilerBase) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
-	log := r.log.WithValues("DataVolume", req.NamespacedName)
-	res, syncErr := r.sync(log, req, nil, nil)
-	return r.updateStatus(log, res, syncErr)
-}
-
-func (r ReconcilerBase) sync(log logr.Logger, req reconcile.Request, cleanup, prepare func(*cdiv1.DataVolume) error) (dataVolumeSyncResult, error) {
-	var ctx dataVolumeSyncResult
-
+func (r ReconcilerBase) syncCommon(log logr.Logger, req reconcile.Request, syncRes *dataVolumeSyncResult, cleanup, prepare func() error) error {
```

Comment:
I think it would be better to have this return `(*dataVolumeSyncResult, error).  The function will create the object, pass it to the callback functions and return it

Reply:
Not sure I agree. The func will need to return the pair instead of a just error (x9 times) which is not that nice (currently the code assumes that dataVolumeSyncResult != nil), and the callbacks will also need a param which is currently not needed. Pleasee convince me why you do prefer it.

Reply:
To me:
```
var syncRes dataVolumeSyncResult
syncErr := r.syncCommon(log, req, &syncRes, nil, nil)
```

is awkward compared to:


```
syncRes, syncErr := r.syncCommon(log, req, nil, nil)
```

IMO it only makes sense to pass a parameter by reference if  1) the parameter is initialized with values or 2) it is being used in some sort of polymorphic way like `json.UnMarshal`.  Neither of those are true here

Also, not that it matters much, I see this function called only three times